sdk/schemaregistry/schema-registry-json/test/public/errors.spec.ts

```diff
@@ -64,211 +65,58 @@ describe("Error scenarios", function () {
-        namespace: "validation",
-        fields: [
-          {
-            name: "name",
-            type: "string",
-          },
-          {
-            name: "age",
-            type: "int",
-          },
-        ],
-      };
-      const message = await serializer.serialize(
-        {
-          name: "",
-          favoriteNumber: 1,
-        },
-        JSON.stringify(writerSchema)
-      );
-      await assertError(
-        serializer.deserialize(message, {
-          schema: JSON.stringify(incompatibleReaderSchema),
-        }),
-        {
-          causeMessage: /no matching field for default-less validation.JsonUser.age/,
-        }
-      );
-    });
-    it("invalid writer schema at time of deserializing", async function (this: Context) {
+    it("invalid schema at time of deserializing", async function (this: Context) {
       /**
        * This test can not run in live mode because the service will validate the schema.
        */
       if (isLiveMode()) {
         this.skip();
       }
       const { id } = await registry.registerSchema({
-        definition: "",
+        definition: testSchema,
         format: "json",
         groupName: testGroup,
         name: "test",
       });
-      const writerSchema = {
-        type: "record",
-        name: "JsonUser",
-        namespace: "validation",
-        fields: [
-          {
-            name: "name",
-            type: "string",
-          },
-          {
-            name: "favoriteNumber",
-            type: "int",
-          },
-        ],
-      };
       const { data } = await serializer.serialize(
         {
           name: "",
           favoriteNumber: 1,
         },
-        JSON.stringify(writerSchema)
-      );
-      await assertError(
-        serializer.deserialize({
-          data,
-          contentType: `json/binary+${id}`,
-        }),
-        {
-          causeMessage: /Unexpected end of JSON input/,
-        }
+        testSchema
       );
-    });
-    it("incompatible writer schema", async function () {
-      const writerSchema1 = {
-        type: "record",
-        name: "JsonUser",
-        namespace: "validation",
-        fields: [
-          {
-            name: "name",
-            type: "string",
-          },
-          {
-            name: "favoriteNumber",
-            type: "int",
-          },
-        ],
-      };
-      const writerSchema2 = {
-        type: "record",
-        name: "JsonUser",
-        namespace: "validation",
-        fields: [
-          {
-            name: "name",
-            type: "string",
-          },
-          {
-            name: "description",
-            type: "string",
-          },
-        ],
-      };
-      const { id } = await registry.registerSchema({
-        definition: JSON.stringify(writerSchema2),
-        format: "json",
-        groupName: testGroup,
-        name: "test",
+      serializer.deserialize({
+        data,
+        contentType: `application/json+${id}`,
```

Comment:
nitpick: Perhaps factor this out into a function so it is easier to edit in the future?

Reply:
I will mark this back as unresolved because I don't see a function to build the content type.